exec/vm.go

```diff
@@ -397,3 +399,49 @@ outer:
 	}
 	return 0
 }
+
+// VMInterface is a proxy passed to host functions in order to access
+// things such as memory and control.
+type VMInterface struct {
+	vm *VM
+}
+
+// NewVMInterface creates a VM interface object for host functions
+func NewVMInterface(vm *VM) *VMInterface {
+	return &VMInterface{vm: vm}
+}
+
+// WriteBytes takes an array of bytes and copies it at offset into the
+// module's memory.
+func (vi *VMInterface) WriteBytes(offset int, data []byte) error {
```

Comment:
this method really looks like an `io.WriterAt`:
```go
type WriterAt interface {
	WriteAt(p []byte, off int64) (n int, err error)
}
```

it feels really tempting to implement it... :)